### PR TITLE
Adding controller to add a sec token annotation

### PR DESCRIPTION
**Pull request by nathan-jenan-rancher**
This change adds a controller that looks for the `sessionToken` to be present
on the EKS config, and if it is there, add an annotation to the cluster.  The
reason this is necessary is becuase credentials that have a `sessionToken`
will expire and the UI must take additional steps to refresh the credentials
before attempting an operation on the cluster, but because `sessionToken` is a
password field, it will be hidden from the UI also.
    
Issue:
https://github.com/rancher/rancher/issues/16005

------------

Also contains a change that modifies how some of the kontainer-engine validation tests were
running to make sure they wait for the `provisioned` condition to appear on
the cluster before asserting that a value is present.  Before this change a
race condition was present and sometimes the tests would fail unnecessarily.

### CODE REVIEW

**Comment by StrongMonkey** [comment]
LGTM

File: pkg/controllers/management/clusterstatus/temporarycredentials.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+	management.Management.Clusters("").AddHandler("temporary-credentials", c.sync)
+}
+
+type clusterAnnotations struct {
+	clusters v3.ClusterInterface
+}
+
+func (cd *clusterAnnotations) sync(key string, cluster *v3.Cluster) error {
+	if key == "" || cluster == nil {
```
Comment (alena1108) [inline pkg/controllers/management/clusterstatus/temporarycredentials.go]: should check a deletion timestamp on a cluster, and return if not nil

File: pkg/controllers/management/clusterstatus/temporarycredentials.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		return nil
+	}
+
+	original := cluster
+	cluster = original.DeepCopy()
```
Comment (alena1108) [inline pkg/controllers/management/clusterstatus/temporarycredentials.go]: can we do a deep copy only when we decide to update? would save us on performance

File: pkg/controllers/management/clusterstatus/temporarycredentials.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	if eksConfig := cluster.Spec.AmazonElasticContainerServiceConfig; eksConfig != nil {
+		newValue := strconv.FormatBool(eksConfig.SessionToken != "")
+
+		if newValue != cluster.Annotations[temporaryCredentialsAnnotationKey] {
```
Comment (alena1108) [inline pkg/controllers/management/clusterstatus/temporarycredentials.go]: Nit: any chance annotations could be nil?

Comment (nathan-jenan-rancher) [reply pkg/controllers/management/clusterstatus/temporarycredentials.go]: Added an initializer for the annotations below this line of code.

**Comment by alena1108** [comment]
LGTM

**Pull request merged**